src/db_queries/expenses.ts

```diff
@@ -0,0 +1,120 @@
+import type CenterExpenses from "../models/center_expenses.model";
+import { Helpers } from '../utils/helpers';
+import { Request, Response } from 'express';
+import dbClient from "../db/client";
+import { QueryResult } from "pg";
+
+const helpers = new Helpers();
+
+export class Expenses {
+
+    private readExpenseQuery = "SELECT * FROM organizations_centers_expenses WHERE center_id = $1";
+    private createExpenseQuery = "INSERT INTO organizations_centers_expenses (center_id, month, year, expense, amount, is_recurring) VALUES ($1, $2, $3, $4, $5, $6)";
+    private deleteExpenseQuery = "DELETE FROM organizations_centers_expenses WHERE center_id = $1 AND id = $2";
+    private updateExpenseQuery = "UPDATE organizations_centers_expenses SET month = $1, year = $2, expense = $3, amount = $4, is_recurring = $5 WHERE center_id = $6 AND id = $7";
+    private totalExpenseQuery = "SELECT SUM(amount) FROM organizations_centers_expenses WHERE center_id = $1";
+    private totalExpensePaidQuery = "SELECT SUM(amount) FROM organizations_centers_expenses_paid WHERE center_id = $1";
+
+    constructor() {
+        helpers.autoBind(this);
+    }
+
+    private async getCenterExpenseQuery(centerId: string): Promise<QueryResult<CenterExpenses>> {
+        const query = this.readExpenseQuery;
```

Comment:
We should keep the query here instead of defining it way above the method because I have used this `node-postgres(pg)` module before. This will start getting messy soon when we begin using complex queries, and the way we pass the variables to the query is in the array, so we need to ensure all the values are in the right order. At that moment, it will be easier to have both query and variables in single-eye view instead of scrolling up and down.

This is a suggestion for better dev experience!

Reply:
@2yk That is a valid concern. But I don't like the idea of having queries duplicated in the endpoint handlers. How about this approach ? 


```

private updateExpenseQuery = ({
    centerId,
    month,
    year,
    expense,
    amount,
    isRecurring,
    id,
  }: CenterExpenses) =>
 [
    "UPDATE organizations_centers_expenses SET month = $1, year = $2, expense = $3, amount = $4, isRecurring = $5 WHERE center_id = $6 AND id = $7",
    [month, year, expense, amount, isRecurring, centerId, id],
  ];
```

`
  // and then calling it like this 
  
        const [query, values] = this.updateExpenseQuery({
        centerId: center_id,
        month,
        year,
        expense,
        amount,
        isRecurring,
        id: expense_id,
      });
      await dbClient.query(query, values);
`

This will remove duplicity concerns of messing up order of values in array. Object would ensure correct values when calling function.